specification/data-semantic-conventions.md

```diff
@@ -149,3 +179,44 @@ one for sent messages and one for received message. This way we guarantee that
+
+| Attribute name | Notes and examples                                           |
+| :------------- | :----------------------------------------------------------- |
+| `code.func` | The method or function name, or equivalent (usually rightmost part of the code unit's name). |
+| `code.ns` | The "namespace" within which `code.func` is defined. Usually the qualified class or module name, such that `code.ns` + some separator + `code.func` form an unique identifier for the code unit. |
+| `code.file` | The source code file name where the code unit is defined. |
+| `code.basepath` | Optional prefix for `code.file`. If present `code.basepath`/`code.file` must be an absolute path. |
+| `code.lineno` | The line number within `code.file` as an integer. |
+
+[network attributes]: #general-network-connection-attributes
+
+## General network connection attributes
+
+These attributes may be used for any network related operation.
+
+|  Attribute name  |                                 Notes and examples                                 |
+| :--------------- | :--------------------------------------------------------------------------------- |
+| `sock.transport` | Transport protocol used. See note below.                                           |
+| `peer.ip`        | Remote address of the peer (dotted decimal for IPv4 or RFC5952 for IPv6)           |
+| `peer.port`      | Remote port number as an integer. E.g., `80`.                                      |
+| `peer.name`      | Remote hostname or similar, see note below.                                        |
+| `host.ip`        | Like `peer.ip` but for the host IP. Useful in case of a multi-IP host.             |
+| `host.port`      | Like `peer.port` but for the host port.                                            |
+| `host.name`      | Like `peer.name` but for the host name. If known, the name that the client used to connect should be preferred.   |
+
+**peer.name**: For IP-based communication, the name should be the host name that was used to look up the IP adress in `peer.ip` (e.g., `"example.com"` if connecting to an URL `https://example.com/foo`). If that is not available, reverse-lookup of the IP can be used to obtain it. If `sock.transport` is `"unix"` or `"pipe"`, the absolute path to the file representing it should be used. If there is no such file (e.g., anonymous pipe), the name should explicitly be set to the empty string to distinguish it from the case where the name is just unknown or not covered by the instrumentation.
+
+**sock.transport**: The name of the transport layer protocol (or the relevant protocol below the "application protocol"). Should be one of these strings:
```

Comment:
Is `sock.transport` mandatory?

Reply:
This section does not define whether a field is mandatory. Other sections refer to this section, e.g. RPC requires `peer.name` https://github.com/dynatrace-oss-contrib/opentelemetry-specification/pull/1/files#diff-612368d7928796edda0d351261863eceR115. Currently, `sock.transport` isn't required anywhere.

Reply:
As the meaning of `peer.name` differs between unix domain sockets and ip I assume some hint is needed.

Reply:
OK, it would make sense to define e.g. in HTTP: "If `sock.transport` is not specified, it is assumed to be `IP.TCP`" (or make it depend on the flavor of HTTP, e.g. QUIC uses UDP)